api/python/quilt3/packages.py

```diff
@@ -340,6 +341,26 @@ def __init__(self, registry, name, top_hash):
         self.top_hash = top_hash
 
 
+class ManifestJSONDecoder(json.JSONDecoder):
+    """
+    Standard json.JSONDecoder reuses same `str` objects for JSON properties, while doing
+    a single `decode()` call.
```

Comment:
So you mean the standard decoder generates a lot of garbage to be collected? I would say that's *not* reusing str objects, but creating fresh ones.